uiServices/src/testSuiteServices.go

```diff
@@ -84,14 +89,169 @@ func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) boo
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(req.Body)
+
+	if !FilePathExist(testSuitePathDir) {
+		logrus.Error("File path does not exist")
+		rw.WriteHeader(http.StatusNotFound)
+		return
+
+	}
+
+	if !ValidateJsonWithSchema(buf.Bytes(),schemaFile, structType) {
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	testSuite := testStrategies.TestSuite{}
+	err := json.Unmarshal(buf.Bytes(), &testSuite)
+
+	if err != nil {
+		logrus.Error("Cannot Unmarshall Json")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !testSuiteWriterXml(testSuite, testSuitePathDir) {
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusNoContent)
+}
+
+func getTestSuite(rw http.ResponseWriter, req *http.Request){
+
+	testSuitePathDir := getTestSuiteHeader(req)
+	testSuiteName := chi.URLParam(req, "testSuiteName")
+
+	ValidateFileNameAndHeader(rw,req,testSuitePathDir, testSuiteName)
+
+	file, err := os.Open(fmt.Sprintf("%s%s.xml", testSuitePathDir, testSuiteName))
+	if err != nil {
+		logrus.Error("Test Suite Name Not Found: "+testSuitePathDir + testSuiteName)
+		rw.WriteHeader(http.StatusNotFound)
+		return
+	}
+	defer file.Close()
+
+
+	var testSuite testStrategies.TestSuite
+
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Read File")
+		return
+	}
+
+	err = xml.Unmarshal(byteValue, &testSuite)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Unmarshall")
+		return
+	}
+
+	testSuiteJSON, err := json.MarshalIndent(testSuite,"","")
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Marshall")
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+	rw.Write(testSuiteJSON)
+	logrus.Println(string(testSuiteJSON))
+
+}
+
+func getAllTestSuites(rw http.ResponseWriter, req *http.Request){
+	var testSuite testStrategies.TestSuite
+	var filename string
+
+	type Suite struct {
+		File string `json:"file"`
+		Name string `json:"name"`
+		Description string `json:"description"`
+	}
+
+	var suite Suite
+
+	type Suites[]Suite
+
+	suites := Suites{}
+
+	testSuitePathDir := getTestSuiteHeader(req)
+	if len(testSuitePathDir) <= 1 {
+		logrus.Error("No file directory entered")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	files, err := ioutil.ReadDir(testSuitePathDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, file := range files {
+		if filepath.Ext(testSuitePathDir + file.Name()) == ".xml" {
+			filename = file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testSuitePathDir, filename))
+			if err != nil {
+				logrus.Error("Cannot open file: ", filename)
+			}
+
+			byteValue, err := ioutil.ReadAll(file)
+			if err != nil{
+				logrus.Error("Cannot Read File: ", filename)
+			}
+
+			err = xml.Unmarshal(byteValue, &testSuite)
+			if err != nil {
+				logrus.Error("Cannot Unmarshall: ", filename)
+			}
+
+			suite.File = filename
```

Comment:
I would change the approach slightly to use local variables:

`suite: Suite{
    Name: testSuite.Name,
    Description: testSuite.Description,
    File: filename,
}

`

Then you can remove the var suite Suite outside the for loop.

You should also be able to remove the resets below because an xml.unmarshal should clear the values? (was this not happening?)

Reply:
I need the resets in it as error handling kind of. Resetting values is not needed if the directory only has Test Suite xml files.  

But I tried GET ALL on a directory with Test Suite files AND config files. 

- If the first file viewed is an Config File, nothing will return. Which is correct 
-If a Config File is read AFTER a Test Suite file is read, it returns a duplicate of the previous Test Suite xml file, despite the fact it reaches the error handling and logs correctly that the Config xml file doesn't not marshall correctly into a Test Suite
- Resetting values after each iteration resolves this problem though

Reply:
I'll pull this down and have a look. It's a little bit hacky as it is.

Reply:
Cool, also if you pull it down, I didn't actually psuh that very last change of :